proposed/0006-Launcher.md

```diff
@@ -0,0 +1,192 @@
+```
+[
+    {
+        App Name,
+        Random Unique App ID,    // 64 Bytes
+        Vec<SHA512(App-Binary)>, // Also serves to tell how many machines where the App is added to Launcher
+        <APP-ROOT_DIR>
+        (PublicKeys (sign and encrypt),
+         PrivateKey (sign and decrypt)),
+        OtherMetadata,
+    }
+    {
+        App Name,
+        Random Unique App ID,    // 64 Bytes
+        Vec<SHA512(App-Binary)>, // Also serves to tell how many machines where the App is added to Launcher
+        <APP-ROOT_DIR>
+        (PublicKeys (sign and encrypt),
+         PrivateKey (sign and decrypt)),
+        OtherMetadata,
+    },
+    … etc
+]
+```
+- Create/Append to local config file (on the user's machine) the following:
+```
+[
+    { App-ID, “/path/to/XYZ” }, // This is newly added by the above
+    { App-ID, “/path/to/some/other/app/already/there” },
+    ... etc.,
+]
+```
+- The format of the config file will be CBOR (compact-binary-object-representation). The name of the local config file should be `<LOCAL-CONFIG-FILE> = .launcher-local-config-file`. The config file location flowchart shall be same as that of `crust` crate's.
+
+**step 2:** User activates the app (eg., double click) from within the Launcher.
+
+**step 3:** Launcher checks the App-ID, reads the path from the `<LOCAL-CONFIG-FILE>` that it made and starts the app as an independent process. The Launcher supplies a random port on which it will listen to this app via command line options.
+
+**./path/to/XYZ --launcher “port:33000;protocol:udp;nonce:random-u32-number”**
+
+**step 4:** Launcher will wait for a predefined time of **15** seconds for data reception on that port. If it times out it will close the socket (release its binding to it)
+
+**step 5:** App responds on the socket asking for Launcher to give keys and its root directory, which Launcher had reserved as `XYZ-Root-Dir`
+- The payload format for this request shall be CBOR encoded sturcture of the following:
+```
+struct Request {
+    prefix: u8,  // This must be ASCII of '?' (a question mark)
+    nonce : u32, // This must be the nonce supplied by Launcher
+}
+```
+
+**step 6:** Launcher reads the `<LAUNCHER-CONFIG-FILE>` and creates a mapping in the MaidManagers for associating user’s `MAID Keys <-> App specific Keys` to allow App to PUT/POST on behalf of the user in `<APP-ROOT-DIR>`.
+- The mapping is done only for the ownership key/s (not encryption keys).
+- The request for mapping/un-mapping shall be a command to the `MaidManagers`. For commands to `MaidManagers`, `StructuredData` with special Type-Tag will be reserved. On reception of this `StructuredData` the vaults will check the payload and act upon the request instead of executing a normal reaction to the usual `PUT/DELETES`.
```

Comment:
You're correct, but Vaults also don't currently handle any special cases of PUT/DELETE either, so there's changes to be made either way.

However, I guess your point is more that my suggestion would impact Routing as well as Vaults?  I do feel that using a PUT here is slightly in conflict with how I saw the network working (I'd anticipated "special actions" to be sent via POST messages), but I'm probably viewing it wrong again, and we can definitely make it work at Vaults as you suggested.